### PR TITLE
Various fixes for dask.distributed

### PR DESCRIPTION
**Pull request by mrocklin**
1.  Extend timeout in tests from 10 to 100 seconds
2.  Sleep until all workers report in in tests accordingly
3.  Workers send pid to scheduler

cc @cowlicks

### CODE REVIEW

**Comment by mrocklin** [comment]
I restarted travis.ci four times and then ran into an intermittent error.

**Comment by mrocklin** [comment]
@cowlicks thoughts on recent changes to the ipcluster test?  I wanted to mark the whole process as slow so I turned the setup-teardown into a context manager

**Comment by mrocklin** [comment]
This is obviously bad if we have many of these tests though, for the moment, significantly decreases test times (a strong motivator for me)

**Comment by mrocklin** [comment]
Still haven't solved the occasional segfault in test_status_client.  I might merge this soon anyway if there aren't any comments by the end of the day.

File: dask/distributed/tests/test_distributed.py [hunk 1]
```diff
@@ -46,31 +47,40 @@ def teardown_module():
...
-        assert_array_almost_equal(result1, a)
-
+        c = Client()
+        yield c
     finally:
-        # close the workers
-        dask_client.close(close_scheduler=True)
+        teardown_cluster()
+
+@pytest.mark.slow
```
Comment (cowlicks) [inline dask/distributed/tests/test_distributed.py]: looks good

File: dask/distributed/tests/test_distributed.py [hunk 2]
```diff
@@ -46,31 +47,40 @@ def teardown_module():
...
+            d3 = da.ghost.trim_internal(d2, {0: 1, 1: 1})
+            result1 = d3.compute(get=dask_client.get)
+            assert_array_almost_equal(result1, a)
+
+        finally:
+            # close the workers
+            dask_client.close(close_scheduler=True)
 
 
 def test_gh_248():
```
Comment (cowlicks) [inline dask/distributed/tests/test_distributed.py]: This test is supposed to be run with the distributed scheduler, but it doesn't look like that was happening before.

Out of scope: we should have a context manager which sets up a local distributed cluster using multiprocessing (not ipython) to run tests like this.

File: dask/distributed/worker.py [hunk 3]
```diff
@@ -110,7 +111,7 @@ def __init__(self, scheduler, data=None, nthreads=100,
 
         self.to_scheduler.setsockopt(zmq.IDENTITY, self.address)
         self.to_scheduler.connect(scheduler)
-        self.send_to_scheduler({'function': 'register'}, {})
+        self.send_to_scheduler({'function': 'register'}, {'pid': os.getpid()})
```
Comment (cowlicks) [inline dask/distributed/worker.py]: :+1:

**Pull request merged**